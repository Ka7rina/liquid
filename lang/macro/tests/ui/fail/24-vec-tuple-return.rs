use liquid_lang as liquid;

#[liquid::contract(version = "0.1.0")]
mod noop {
    #[liquid(storage)]
    struct Noop {}

    #[liquid(methods)]
    impl Noop {
        pub fn new(&mut self) {}

        pub fn noop_0(&self) -> (bool, i32) {}

        pub fn noop_1(&self) -> Vec<(bool, i32)> {}
    }
}

fn main() {}
