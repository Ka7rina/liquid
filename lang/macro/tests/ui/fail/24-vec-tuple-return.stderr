error[E0277]: the trait bound `(bool, i32): You_Should_Use_An_Valid_Parameter_Type` is not satisfied
  --> $DIR/24-vec-tuple-return.rs:14:30
   |
14 |         pub fn noop_1(&self) -> Vec<(bool, i32)> {}
   |                              ^^^^^^^^^^^^^^^^^^^ the trait `You_Should_Use_An_Valid_Parameter_Type` is not implemented for `(bool, i32)`
   |
   = note: required because of the requirements on the impl of `You_Should_Use_An_Valid_Return_Type` for `std::vec::Vec<(bool, i32)>`

error[E0277]: the trait bound `(bool, i32): MediateEncode` is not satisfied
  --> $DIR/24-vec-tuple-return.rs:14:30
   |
14 |         pub fn noop_1(&self) -> Vec<(bool, i32)> {}
   |                              ^^^^^^^^^^^^^^^^^^^ the trait `MediateEncode` is not implemented for `(bool, i32)`
   |
  ::: $WORKSPACE/lang/src/traits.rs:24:18
   |
24 |     type Output: Encode + 'static;
   |                  ------ required by this bound in `FnOutput`
   |
   = note: required because of the requirements on the impl of `MediateEncode` for `std::vec::Vec<(bool, i32)>`
   = note: required because of the requirements on the impl of `Encode` for `std::vec::Vec<(bool, i32)>`
